DigicielFull/force-app/main/default/classes/InfolegaleOpp_BAT.cls

```diff
@@ -13,10 +12,20 @@ global class InfolegaleOpp_BAT implements Database.Batchable<sObject>, Database.
         setIdOpp = strParam;
     }
 
+    /**
+     * Method to retrieve opportunties from database if present on the setIdOpp
+     * @param bc  BatchableContext object for executing batch job
+     * @return query locator for retrieved opportunties
+     */
     global Database.QueryLocator start(Database.BatchableContext bc) {    
         return Database.getQueryLocator([SELECT Id, INFOLEGALERating__c, Score_INFOLEGALE__c, AccountId, StageName, Account.Owner.Digicel_Country__c FROM Opportunity WHERE Id IN :setIdOpp AND Account.Owner.Digicel_Country__c = 'FWI' ]);            
     }
 
+    /**
+     * method to update the INFOLEGALERating__c and Score_INFOLEGALE__c field on the Opportunity records.
+     * @param bc Database.BatchableContext - context of the batch job
+     * @param listOpps List<Opportunity> - list of opportunties to be updated
+     */
     global void execute(Database.BatchableContext bc, List<Opportunity> listOpps){
         List<Opportunity> listOppToUpdate = new List<Opportunity>();
         List<Account> accToUpdate = new List<Account>();
```

Comment:
@PillowR ligne 36 tu fais une requête sur l'object account, tu pourrais directement les inclure dans la requête de ton start :
[SELECT Id, INFOLEGALERating__c, Score_INFOLEGALE__c, AccountId, StageName, Account.Owner.Digicel_Country__c FROM Opportunity WHERE Id IN :setIdOpp AND Account.Owner.Digicel_Country__c = 'FWI' ] 
->
[SELECT Id, INFOLEGALERating__c, Score_INFOLEGALE__c, AccountId, StageName, Account.Owner.Digicel_Country__c, Account.INFOLEGALERating__c, account.Registration_Number__c FROM Opportunity WHERE Id IN :setIdOpp AND Account.Owner.Digicel_Country__c = 'FWI' ]